apis/kueue/webhooks/clusterqueue_webhook.go

```diff
@@ -30,57 +31,70 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
 )
 
 var (
 	// log is for logging in this package.
 	clusterQueueLog = ctrl.Log.WithName("cluster-queue-webhook")
 
-	queueingStrategies = sets.NewString(string(StrictFIFO), string(BestEffortFIFO))
+	queueingStrategies = sets.NewString(string(kueue.StrictFIFO), string(kueue.BestEffortFIFO))
+)
+
+const (
+	isNegativeErrorMsg string = `must be greater than or equal to 0`
 )
 
-const isNegativeErrorMsg string = `must be greater than or equal to 0`
+type ClusterQueueWebhook struct{}
 
-func (cq *ClusterQueue) SetupWebhookWithManager(mgr ctrl.Manager) error {
+func setupWebhookForClusterQueue(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(cq).
+		For(&kueue.ClusterQueue{}).
+		WithDefaulter(&ClusterQueueWebhook{}).
+		WithValidator(&ClusterQueueWebhook{}).
 		Complete()
 }
 
 // +kubebuilder:webhook:path=/mutate-kueue-x-k8s-io-v1alpha1-clusterqueue,mutating=true,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=clusterqueues,verbs=create,versions=v1alpha1,name=mclusterqueue.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Defaulter = &ClusterQueue{}
+var _ webhook.CustomDefaulter = &ClusterQueueWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *ClusterQueueWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	cq := obj.(*kueue.ClusterQueue)
 
-// Default implements webhook.Defaulter so a webhook will be registered for the type
-func (cq *ClusterQueue) Default() {
 	clusterQueueLog.Info("defaulter", "clusterQueue", klog.KObj(cq))
```

Comment:
Let's make this V(5), like the others. Also, the message should be a sentence.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#remove-string-formatting-from-log-message

Reply:
> Let's make this V(5), like the others. Also, the message should be a sentence.

Should we? V(5) is for tracing, maybe V2 or V3, see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#what-method-to-use